### ISSUE TITLE
XML not properly encoded

### ISSUE THREAD

**Issue by cpfeiffer** [TOPdesk/dart-junitreport#26]
It looks like the input characters are not properly XML-encoded.
E.g. when we get this message from `flutter test`:
```
{
  "testID": 4,
  "messageType": "print",
  "message": "\u001b[38;5;196m⛔  2021-03-26 15:48:38.982 MockArticleData - Cannot load promotions\u001b[0m",
  "type": "print",
  "time": 2357
}
```
tojunit produces this illegal XML:
```
<system-out>[38;5;196mâ›”  2021-03-26 15:14:50.656 MockArticleData - Cannot load promotions[0m
....
</system-out>
```
The `\u001b` should be encoded.

**Comment by cpfeiffer** [comment]
It looks like there's even a patch: https://github.com/Tokenyet/dart-junitreport/commit/278665e85f8f7ba630e0edcc47577e7506db7c9d

**Comment by rspilker** [comment]
That should be a problem addressed in the XML library. I'll see if I can still reproduce it.

Can someone upload a file that currently goes wrong?

**Comment by rspilker** [comment]
See also https://github.com/renggli/dart-xml/issues/123

After that has been resolved, most changed from 09820c52c9c39c825567864b73d44402bc742b1b can be reverted.
